### ISSUE TITLE
osxaudio should support Voice-Processing IO

### ISSUE THREAD

**Issue by ikonst** [EricssonResearch/openwebrtc#153]
This is a GStreamer issue documented here:
https://bugzilla.gnome.org/show_bug.cgi?id=732896

I'm adding it here since I found out WebRTC switched from Remote IO to Voice-Processing IO two years ago, and I suspect we should too:
https://webrtc-codereview.appspot.com/1061007

Currently osxaudio is a mess of HAL and Remote IO implementations. IMHO it should be refactored to have selectable implementations:
* remoteio
* vpio
* hal (OS X only)

**Comment by superdump** [comment]
Yup, I was aware of that gst issue but thanks for noting it here. This forms part of the echo cancellation feature in the roadmap. :smile:
